Gems/AWSCore/Code/Source/AWSCoreSystemComponent.cpp

```diff
@@ -137,12 +137,21 @@ namespace AWSCore
 
     void AWSCoreSystemComponent::InitAWSApi()
     {
-        AWSNativeSDKInit::InitializationManager::InitAwsApi();
+        // Multiple different Gems might try to use the AWSNativeSDK, so make sure it only gets initialized / shutdown once
+        // by the first Gem to try using it.
+        m_ownsAwsNativeInitialization = !AWSNativeSDKInit::InitializationManager::IsInitialized();
+        if (m_ownsAwsNativeInitialization)
+        {
+            AWSNativeSDKInit::InitializationManager::InitAwsApi();
+        }
```

Comment:
This seems like a good candidate to store the InitializationManager inside of an `AZ::Interface`

The [AZ::Interface::Register](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Interface/Interface.h#L71-L74) function will not register an interface that is already registered
```c++
// In AWSCoreSystemComponent.h
AZStd::unique_ptr<AWSNativeSDKInit::InitializationManager> m_awsInitializationManager = AZStd::make_unique<AWSNativeSDKInit::InitializationManager>();
// ...
// In AWSCoreSystemComponent.cpp InitAWSApi
if(auto registerOutcome = AZ::Interface<AWSNativeSDKInit::InitializationManager>::Register(m_awsInitializationManager.get());
    !registerOutcome)
{
    AZ_Warning("AWS Core", "AWS Native SDK initialization has failed with error %s", registerOutcome.GetError().c_str());
}

// In AWSCoreSystemComponent.cpp ShutdownAWSApi
// Unregister will only unregister an the instance if the memory address matches
// otherwise a failure outcome will be returned.
// For this use case the failure result is needed
AZ::Interface<AWSNativeSDKInit::InitializationManager>::Unregister(m_awsInitializationManager.get());

```

The [AWSNativeSDKInit.h](https://github.com/o3de/o3de/blob/development/Code/Tools/AWSNativeSDKInit/include/AWSNativeSDKInit/AWSNativeSDKInit.h#L28-L30) InitializationManager class could just add a use of the `AZ_RTTI` macro and then it would be able to be stored in an AZ::Interface

Reply:
In this case, I'd be inclined to leave it as-is. The InitializationManager is literally only used to call InitAwsApi() / Shutdown(), so having it available through an Interface doesn't provide any additional value. Using the Interface to prevent multiple initializations just seems to me like a slightly more obfuscated way to accomplish the same goal. I'm not _opposed_ to changing it if you feel strongly though, I only have a small preference for this one.

Reply:
I am cool with leaving this as is.